src/Grid/topology.jl

```diff
@@ -86,11 +135,22 @@ function _get_facet_facet_neighborhood(::ExclusiveTopology, #=rdim=#::Val{:mixed
     Access the `vertex_vertex_neighbor`, `edge_edge_neighbor`, or `face_face_neighbor` fields explicitly instead."))
 end
 
-function Base.show(io::IO, ::MIME"text/plain", topology::ExclusiveTopology)
-    println(io, "ExclusiveTopology\n")
-    print(io, "  Vertex neighbors: $(size(topology.vertex_vertex_neighbor))\n")
-    print(io, "  Face neighbors: $(size(topology.face_face_neighbor))\n")
-    println(io, "  Edge neighbors: $(size(topology.edge_edge_neighbor))")
+# Guess of how many neighbors depending on grid dimension and index type.
+# This is just a performance optimization, and a good default is sufficient.
+_getsizehint(::AbstractGrid{3}, ::Type{FaceIndex}) = 1 # 2
+_getsizehint(::AbstractGrid, ::Type{FaceIndex}) = 0 # No faces exists in 2d or lower dim
+_getsizehint(::AbstractGrid{dim}, ::Type{EdgeIndex}) where dim = 1 #dim^2
+_getsizehint(::AbstractGrid{dim}, ::Type{VertexIndex}) where dim = 1 # 2^dim
+_getsizehint(::AbstractGrid{1}, ::Type{CellIndex}) = 2
+_getsizehint(::AbstractGrid{2}, ::Type{CellIndex}) = 12
```

Comment:
Maybe we should add an additional layer to also dispatch on the cell type to not overestimate too much. Where do you have the numbers from?

Reply:
Yes, the commented versions are from hypercubes if I recall correctly, but an overestmate for the exclusive topology. 

Not sure if it is worthwhile to make this general since it isn't super-sensitive in my benchmarks (of course it matters, but now allocations are a quite small fraction of the time). So could even be better to simplify further and hard-code in `ExclusiveTopology`, since the same sizehints won't be valid for e.g. `CoverTopology` (didn't think of this when starting to implement this)

Reply:
Posting a quick function to get some size info for later use
```julia
function analyze_top_sizes(grid)
    top = ExclusiveTopology(grid);
    for key in fieldnames(typeof(top))
        nbh = getfield(top, key)
        nbh isa Ferrite.ArrayOfVectorViews || continue
        max_size = maximum(length, nbh; init=0)
        avg_size = sum(length, nbh; init=0)/length(nbh)
        println(key)
        println("  max: ", max_size, " entities")
        println("  avg: ", avg_size, " entities")
    end
end
```